src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -430,14 +430,20 @@ public async Task<IActionResult> GetTriggers()
         {
             _metricsLogger.LogEvent(MetricEventNames.GetTriggersInvoked);
 
-            var result = await _functionsSyncManager.GetTriggersAsync();
+            // This endpoint is only available in validation mode
+            if (_environment.IsInValidationMode())
+            {
+                var result = await _functionsSyncManager.GetTriggersAsync();
 
-            // GetTriggersAsync() does not swallow exceptions,so any failures will still be a 500.
-            // The only result.Success == false we need to consideris when the environment
-            // does not support sync triggers.
-            return result.Success
-                ? Ok(result)
-                : StatusCode(StatusCodes.Status403Forbidden, new { status = result.Error });
+                // GetTriggersAsync() does not swallow exceptions, so any failures will still be a 500.
+                // The only result.Success == false we need to consider is when the environment
+                // does not support sync triggers.
+                return result.Success
+                    ? Ok(result)
+                    : StatusCode(StatusCodes.Status403Forbidden, new { status = result.Error });
+            }
+
+            return StatusCode(StatusCodes.Status403Forbidden, new { status = "GetTriggers operation is Forbidden." });
```

Comment:
Can we have a more description error message? Just so a caller won't think it is a permission related issue.

```suggestion
            return StatusCode(StatusCodes.Status403Forbidden, new { status = "GetTriggers operation is not supported in for this instance." });
```

Reply:
Updated status text :)